packages/protocol/scripts/create-stable-token.ts

```diff
@@ -0,0 +1,265 @@
+} from '@celo/protocol/lib/web3-utils'
+import { config } from '@celo/protocol/migrationsConfig'
+import { toFixed } from '@celo/utils/lib/fixidity'
+import { Exchange{TICKER}Instance, FreezerInstance, ReserveInstance } from 'types'
+
+const initializeArgs = async (): Promise<any[]> => {
+  return [
+    config.registry.predeployedProxyAddress,
+    CeloContractName.StableToken{TICKER},
+    toFixed(config.exchange.spread).toString(),
+    toFixed(config.exchange.reserveFraction).toString(),
+    config.exchange.updateFrequency,
+    config.exchange.minimumReports,
+  ]
+}
+
+module.exports = deploymentForCoreContract<Exchange{TICKER}Instance>(
+  web3,
+  artifacts,
+  CeloContractName.Exchange{TICKER},
+  initializeArgs,
+  async (exchange: Exchange{TICKER}Instance) => {
+    if (config.exchange.frozen) {
+      const freezer: FreezerInstance = await getDeployedProxiedContract<FreezerInstance>(
+        'Freezer',
+        artifacts
+      )
+      await freezer.freeze(exchange.address)
+    }
+
+    const reserve: ReserveInstance = await getDeployedProxiedContract<ReserveInstance>(
+      'Reserve',
+      artifacts
+    )
+    // cUSD doesn't need to be added as it is currently harcoded in Reserve.sol
+    await reserve.addExchangeSpender(exchange.address)
+    await exchange.activateStable()
+  }
+)
+`
+
+function errorFunct(err) {
+  if (err) return console.log(err)
+}
+
+try {
+  const argv = require('minimist')(process.argv.slice(2), {
+    string: ['stableTokenTicker'],
+  })
+  const fiatTicker = argv.stableTokenTicker
+
+  const stabilityContractPath = './contracts/stability'
+  const stabilityProxyPath = './contracts/stability/proxies'
+  const migrationPath = './migrations'
+
+  // contracts
+  fs.writeFile(
+    `${stabilityContractPath}/StableToken${fiatTicker}.sol`,
+    StableTokenTemplate.replace(KEY_TO_BE_REPLACED_IN_TEMPLATES, fiatTicker),
+    errorFunct
+  )
+  fs.writeFile(
+    `${stabilityContractPath}/Exchange${fiatTicker}.sol`,
+    ExchangeTemplate.replace(KEY_TO_BE_REPLACED_IN_TEMPLATES, fiatTicker),
+    errorFunct
+  )
+
+  // proxy
+  fs.writeFile(
+    `${stabilityProxyPath}/StableToken${fiatTicker}.sol`,
+    StableTokenProxyTemplate.replace(KEY_TO_BE_REPLACED_IN_TEMPLATES, fiatTicker),
+    errorFunct
+  )
+  fs.writeFile(
+    `${stabilityProxyPath}/Exchange${fiatTicker}.sol`,
+    ExchangeProxyTemplate.replace(KEY_TO_BE_REPLACED_IN_TEMPLATES, fiatTicker),
+    errorFunct
+  )
+
+  // migration
+  fs.writeFile(
+    `${migrationPath}/09_999_stableToken_${fiatTicker}.sol`,
+    migrationStableTemplate.replace(KEY_TO_BE_REPLACED_IN_TEMPLATES, fiatTicker),
+    errorFunct
+  )
+  fs.writeFile(
+    `${migrationPath}/10_999_exchange_${fiatTicker}.sol`,
+    migrationExchangeTemplate.replace(KEY_TO_BE_REPLACED_IN_TEMPLATES, fiatTicker),
+    errorFunct
+  )
+
+  console.log(`Other thinks that should be updated:
+  * Add constitution parameters: packages/protocol/governanceConstitution.js
```

Comment:
Could we generate a copy pastable block that should be inserted there?

Reply:
Done.